### ISSUE TITLE
Not compatible with IE 11

### ISSUE THREAD

**Issue by natheeshsunway** [ankeetmaini/react-infinite-scroll-component#276]
This library not compitable with ie 11, While scrolling, methods are not called

**Comment by clytras** [comment]
@natheeshsunway why do you think such a library/component should have support for IE11 in 2021 ?

**Comment by natheeshsunway** [comment]
Still some clients need compaitablity of IE 11

**Comment by clytras** [comment]
@natheeshsunway so you believe that because people may still have **some** clients that deny or can't upgrade their browsers, maintainers should spend a significant amount of their time to make this library compatible with an obsolete browser and spend even more hours in the future to keep this support with every new feature they may add?

**Comment by iamthe-Wraith** [comment]
@natheeshsunway you did the right thing. you reported a bug...regardless if people like @clytras (who isn't even a contributor, btw) think it should be resolved, you did exactly what you should do...report it and let the people actually doing the work decide if they will solve it or not.

**Comment by clytras** [comment]
@iamthe-Wraith how this is being considered **a BUG**? Waiting for you to explain and maybe be more specific.

I have an old **Windows 2000** computer that sits on the basement and it has **Internet Explorer 6** installed... I will start requesting from people randomly to support IE 6 because hey, I'm still using it...

So before randomly replying to discussion’s **that you have nothing to add**, **THINK** of what the context is **ABOUT** which is something that you failed to do. And the fact that I'm not a contributor is not of your business at all, you aren't either.

And **YES**, people are moving forward and asking for support for an **obsolete browser** that [has less than a year of official support left](https://www.infoq.com/news/2021/05/microsoft-IE11-retired-june-2022/) is like **asking from maintainers to do more work now and in the future to support something that is near the end of its life**.

So, I ask you to be more careful about your statements. This **IS NOT A BUG**, this **IS A REQUEST** that someone has regarding supporting some obsolete software. I never asked this to be resolved, as there is nothing to resolve here, only a problem from the past that should stay exactly there... in the past.

**Comment by iamthe-Wraith** [comment]
@clytras you are absolutely right. my appoligies for using the incorrect term. instead of **bug**, i should have said **request**.

As for the rest of your rant, you should know that I did **THINK** about what I wrote before posting it... I did exactly what I set out to do... To encourage someone who was given an unnecessarily passive aggressive response to their **request** to continue contributing.

There are far better ways to discuss this topic than to make someone feel bad for posting about it. That is the quickest way to get someone to stop contributing...and in an open source community, any form of contribution should be encouraged, whether you personally agree with it or not.

Your points are valid...yes, official support for IE is ending. yes, i'm sure it would take considerable time for the maintainers/contributers to add this kind of support. But 1. that should be a decision made by the people doing the work...not you...not me. and 2. there are far more effective ways to inform someone about this than the way you responded to this issue. So perhaps before you respond with sarcasm the way you did, you should consider what you are actually adding and accomplishing with such a post.

If you are lucky enough to not have to support IE, that is awesome for you, and I am envious. But you should also understand and consider that not everyone is as fortunate. There are going to be people in the world that will need to continue to support it for a while now, and there is absolutely nothing wrong with them making a request for support. The worst that could happen is the answer is "No". But they should still at least make the request, because hey, you never know right!? If enough people join in and request it, it could encourage someone to step up and make the changes for all those people.

**Comment by clytras** [comment]
@iamthe-Wraith apologies accepted. And I'm not ranting about it, I'm just defending my statements maybe using an acute way.

I'm not trying to be passive aggressive here. I never demanded the O/P author to close this nor attacked them on asking support for obsolete software, I'm just trying to make a statement and present the facts and the reality by asking the author some very valid questions in order to make him think that maybe **it doesn't worth it for contributors to spend their time to support obsolete software**. I'm not into any phycological aspects of how people should ask questions (*nor I care about either*) as long as they don't use bad language nor *ad hominem* instead of focusing on the subject (*which I feel you did by implying that since I'm not a contributor, I should not write about a very valid perspective and reality*).

I didn't try to take any decision nor I tried to force the author to close his question, I just asked him some questions and waited to get some answers that I never did. As I stated above, you may use whatever king of written speech style you want; I will use whatever kind of written speech style I like and if I insult anyone (*which I didn't by the way*) there are authorities to report to and judge the content. Also, I don't think it's sarcasm to ask valid question regarding the subject.
I asked:

> so you believe that because people may still have some clients that deny or can't upgrade their browsers, maintainers should spend a significant amount of their time to make this library compatible with an obsolete browser and spend even more hours in the future to keep this support with every new feature they may add?

And you see some sarcasm here and not a valid question? Why, because I asked him what he believes? That's what I want to know, his opinion regarding the perspective I presented to him.

Yes, if I'm working for a company and my boss insists that I should support IE 11 then I'll probably seek ways to do it in order to keep my job, **but asking other people to do it for me on their free time** has nothing to do with OS contribution at all. I have stopped to support IE11 to apps I'm developing and used by thousands a couple of years ago and yes **I still have some contracts that require IE11 and even IE10 support** and I manage to deliver that support **without promoting it nor asking others to do it for me**. Yes, it's very wrong to support obsolete software and everybody should stop doing it so for people to save countless hours trying to make old and new features compatible with obsolete software. You're talking about opening a *"Let's give it a shot"* issue to even spam contributors with requests that are pointless. Yes, it's pointless to spend a significant amount of time to make something work on something that is being ceased and near EOL and I don't have to be a particular contributor to this repo here in order to question people regarding such requests while of course **not asking them to close it**.

People and companies have to learn that they need to upgrade their browsers, which it should require some time and money, but we’re talking about software near EOL here; by asking and creating such support, 1. Takes countless of hours for existing features, 2. Will take a significant amount of time to support new features, 2. Make people using obsolete software feel safe and do nothing about it since their services still working. I hope you’re getting my points here and I also hope you to agree that I don’t have to be a contributor to express them.

**Comment by natheeshsunway** [comment]
I think it's good to list the supported browser in home page/Readme and close this issue.

**Comment by iamdarshshah** [comment]
IE 11 is not a supported browser. Microsoft ended support for IE 11 in June 2022, and v7.1.0 uses `IntersectionObserver` which IE 11 does not support. We will not be adding a polyfill or workaround for this. If you need IE 11 support, this library is not the right fit. Closing as out of scope.
